cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -54,6 +54,65 @@ namespace {
 constexpr char kTfStaticTopic[] = "/tf_static";
 namespace carto = ::cartographer;
 
+std::tuple<carto::mapping::proto::PoseGraph,
+           carto::mapping::proto::AllTrajectoryBuilderOptions>
+CreatePoseGraph(const std::string& pose_graph_filename,
+                const int num_bag_files) {
+  carto::mapping::proto::PoseGraph pose_graph_proto;
+  carto::mapping::proto::AllTrajectoryBuilderOptions
+      all_trajectory_builder_options;
+
+  carto::io::ProtoStreamReader reader(pose_graph_filename);
+  CHECK(reader.ReadProto(&pose_graph_proto));
+  CHECK(reader.ReadProto(&all_trajectory_builder_options));
+  CHECK_EQ(pose_graph_proto.trajectory_size(), num_bag_files)
+      << "Pose graphs contains " << pose_graph_proto.trajectory_size()
+      << " trajectories while " << num_bag_files
+      << " bags were provided. This tool requires one bag for each "
+         "trajectory in the same order as the correponding trajectories in the "
+         "pose graph proto.";
+  return std::make_tuple(pose_graph_proto, all_trajectory_builder_options);
+}
+
+std::unique_ptr<carto::io::PointsProcessorPipelineBuilder>
+CreatePipelineBuilder(
+    const std::vector<carto::mapping::proto::Trajectory>& trajectories,
+    const std::string file_prefix) {
+  const auto file_writer_factory = [file_prefix](const std::string& filename) {
+    return carto::common::make_unique<carto::io::StreamFileWriter>(file_prefix +
+                                                                   filename);
+  };
+  auto builder =
+      carto::common::make_unique<carto::io::PointsProcessorPipelineBuilder>();
+  carto::io::RegisterBuiltInPointsProcessors(trajectories, file_writer_factory,
+                                             builder.get());
+  builder->Register(RosMapWritingPointsProcessor::kConfigurationFileActionName,
+                    [file_writer_factory](
+                        carto::common::LuaParameterDictionary* const dictionary,
+                        carto::io::PointsProcessor* const next)
+                        -> std::unique_ptr<carto::io::PointsProcessor> {
+                      return RosMapWritingPointsProcessor::FromDictionary(
+                          file_writer_factory, dictionary, next);
+                    });
+  return std::move(builder);
```

Comment:
This doesn't need the `std::move`, right?

Reply:
Yes, you're right. Adjusted the other places, where a unique_ptr was returned, as well.